### DISCUSSION TITLE
Buffered events

### DISCUSSION THREAD

**Discussion by elmcapp** [socketio/socket.io#4601] [Q&A] [answered]
Is there a way to get an array or list of Buffered events? Let say that the client disconnections and later reconnections. In this case we would have Buffered events but I would like to see what events are buffered as I may not want to emit those events after some time. Another use case would be that I may want to see why there is a delay or messages being received late. I could see what all message are in cue to be sent.

**Comment by darrachequesne** [comment, marked as answer]
Hi! You can use the `sendBuffer` array:

```js
socket.emit("hello");

console.log(socket.sendBuffer); // prints "[ { type: 2, data: [ 'hello' ], options: { compress: true } } ]"
```

Source: https://github.com/socketio/socket.io-client/blob/a1c528b089773d7810a03befaeb982f7e01c3e11/lib/socket.ts#L174

**Reply by elmcremotebridge** [reply]
Cool. So my next question, are we able to tell socket.io which of the message to send and which ones to dispose?

**Reply by elmcremotebridge** [reply]
Well I guest your link answered the question above. Thanks
